### PR TITLE
Premium Analytics: draw Posting activity as twelve month calendars

### PR DESCRIPTION
**Pull request by chihsuan**
Fixes WOOA7S-2014

## Proposed changes

The Insights **Posting activity** card currently squeezes all posting history into one continuous heatmap, making it hard to read on sites with years of posts.

Replace it with twelve small month calendars and rename to `Monthly posting activity`, similar to the old Stats card:

* Use the site's timezone, start weeks on Monday, and fade future days without showing tooltips.
* Spread the months out on wide cards and let them scroll sideways on narrow cards.
* Show the **Fewer posts / More posts** legend at the bottom, hiding it on one-row cards to save space.
* Fetch up to 3,000 posts so busy sites include more of their posting history.
* Show a matching loading placeholder while the data loads.

This uses the month calendar layout from #52259.  

## Related product discussion/links
* WOOA7S-2014, parent WOOA7S-2008.
* Stacked on #52259; merge that first.

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions

**Setup**

1. Check out this branch and run `jetpack build plugins/premium-analytics --deps` on a site with posts over the last year, Premium Analytics enabled through the filter.
2. Open **Stats v2 → Insights**.

**Month calendars**

1. Find **Monthly posting activity**. Confirm twelve month blocks, the current month last, month names beneath.
2. Filter the Network tab on `stats/streak`. Confirm `startDate` is the first of the month eleven months ago, `endDate` is today in the site's timezone, and `max=3000`.
3. Hover a day with posts. Confirm "N posts" and the full date. Hover a day after today. Confirm it is faded and shows nothing.
4. Tab to the grid and hold ArrowRight. Confirm the selection crosses into the next month and the tooltip follows.

**Card sizes**

1. Click **Customize** and resize the card to two rows. Confirm the legend sits at the bottom with the grid centred above it.
2. Resize it back to one row. Confirm the legend is gone and there is no vertical scrollbar.
3. Narrow the browser until the months no longer fit. Confirm only the grid scrolls sideways and the legend stays put.

Same site, same two-row card, same data; only the card body changes.

| Before | After |
| --- | --- |
| Continuous heatmap | Screenshot 2026-09-17 at 10 30 10 AM |

https://github.com/user-attachments/assets/abb7010a-964f-4c8f-9d9f-19e4283695f2

### CODE REVIEW

File: projects/packages/premium-analytics/packages/widgets-toolkit/src/components/month-calendar-heatmap/month-calendar-heatmap.tsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+		),
+		[ emptyLabel, formatValue ]
+	);
+
+	return (
+		<div className={ styles.root }>
+			<HeatmapChart
+				data={ data }
+				columnGroups={ columnGroups }
+				compact
```
Comment (louwie17) [inline projects/packages/premium-analytics/packages/widgets-toolkit/src/components/month-calendar-heatmap/month-calendar-heatmap.tsx:84]: The charts docs recommend calendar navigation for month calendars, so the arrow keys step through the days instead of across the whole grid:

```suggestion
				compact
				keyboardNavigation="calendar"
```

With it, the first press lands on the first day of the range, ←/→ step one day at a time, running on into the neighbouring month, ↑/↓ move a week across the month edge, and Page Up / Page Down jump a month. Could you also add a widget test that steps across a month boundary? Right now nothing covers the keyboard.

File: projects/packages/premium-analytics/packages/widgets-toolkit/src/helpers/calendar-heatmap-window.ts [hunk 2]
```diff
@@ -104,24 +71,3 @@ export function buildDenseDaySeries(
...
-/**
- * Days of history worth requesting for a calendar heatmap at a given viewport
- * width. Compact-cell capacity stands in for the widest tile's column ceiling,
- * quantized to whole years so resizing can't fire a request per column gained.
- *
- * @param viewportWidth - Viewport width in px.
- * @return Days of history to request.
- */
-export function resolveCalendarHeatmapWindowDays( viewportWidth: number ): number {
-	const capacityDays = compactCalendarHeatmapCapacity( viewportWidth ) * 7;
```
Comment (louwie17) [inline projects/packages/premium-analytics/packages/widgets-toolkit/src/helpers/calendar-heatmap-window.ts:123]: This was the last caller of `compactCalendarHeatmapCapacity`, so that helper is now dead code ([calendar-heatmap-layout.ts#L113](https://github.com/Automattic/jetpack/blob/d64233f358d952abd09690eb8b3d3cb0edab25d0/projects/packages/premium-analytics/packages/widgets-toolkit/src/helpers/calendar-heatmap-layout.ts#L113)). Could you remove it along with `COMPACT_CELL_SIZE` / `COMPACT_CELL_GAP`, its export in `helpers/index.ts`, and its `describe` block in `__tests__/calendar-heatmap-layout.test.ts`?

**Review by louwie17 — CHANGES_REQUESTED** [review]
Thanks @chihsuan! Tested at `d64233f3` on a local Docker site. The month calendars look good:

- **Request:** `stats/streak?startDate=2025-10-01&endDate=2026-09-17&max=3000`, so the first of the month eleven months ago through today in the site's timezone.
- **Calendars:** twelve blocks, Oct through Sep, with the current month last. A busy day shows "1599 posts" with its date, and days after today are faded with no tooltip.
- **One-row card:** the legend is hidden and there is no vertical scrollbar. On a narrow viewport only the grid scrolls sideways.
- No console errors. The jest suites for the changed files pass (13 suites, 188 tests). Codex also reviewed the diff.

Requesting changes for two things (details inline):

1. **Keyboard navigation.** The chart uses the default `grid` navigation, so the arrow keys don't follow the calendar. The first press lands on Oct 6 rather than Oct 1, → jumps from Oct 12 to Nov 3, and ↓ stops at the end of the month. Every day can be reached, but only by working out the grid. `keyboardNavigation="calendar"` fixes this: I tried it locally, and the arrows step day by day and week by week across month edges, this feels much nicer.

   Now (`grid`):

https://github.com/user-attachments/assets/215e27b4-6bd8-4160-abc3-9e5f06711e6b


   With `keyboardNavigation="calendar"`:


https://github.com/user-attachments/assets/c851d142-22cc-48c4-8e0b-279dfaabc75e

2. **Dead code.** `compactCalendarHeatmapCapacity` lost its only caller in this PR.

Non-blocking: when the card is narrower than the twelve months, the grid opens scrolled to the oldest month, so the current month starts off-screen. I think that's expected, but worth deciding on purpose. Scrolling to the end on mount would show the latest month first.

**Review by dognose24 — COMMENTED** [review]
Tested at `d64233f` on a local Docker site (Jetpack-connected, `gmt_offset = -12`). The calendars render as described: `stats/streak?startDate=2025-10-01&endDate=…&max=3000`, twelve blocks with the current month last, weeks from Monday, future days faded with no tooltip, and the legend dropped on the one-row Insights default. Date window, timezone handling, empty-state filtering and the changelog set (jetpack / premium-analytics / wpcomsh, mu-wpcom-plugin skipped) all check out; the suite passes under `TZ=Asia/Tokyo`, `America/Los_Angeles` and `Pacific/Kiritimati`.

+1 to both of @louwie17's open items (`keyboardNavigation="calendar"` and the now-dead `compactCalendarHeatmapCapacity`). A few more, none blocking on their own:

**Initial scroll position**
On a ~1300px viewport the full-width one-row card cannot fit twelve months at the minimum group gap, so the grid scrolls, and the month that ends up off-screen is the current one:


截圖 2026-09-17 晚上11 28 03

The prototype has the same overflow (its two screenshots in WOOA7S-2014 are just scrolled to opposite ends), so this is not a deviation from the design, but since we scroll anyway, could the grid start scrolled to the end so the current month is visible and the oldest ones are what's hidden? A `scrollLeft = scrollWidth` on mount, or a follow-up if you'd rather keep this PR to the layout.

**Period in the title**
The card pins its own window and ignores the date picker, but only the info tooltip says so. #52308 put the period into the leaderboard titles for the same reason; "Monthly posting activity in the last 12 months" would match its "in the last 7 days" phrasing. The help copy can stay as is; the three changelog entries and the test's `ariaLabel` would follow.

**Tests**
- `posting-activity.test.tsx` L94–98 re-asserts the Monday layout (`data-column` on Oct 1) with the same comment as the component test. The component test owns that; the widget test can drop both.
- `widgets/posting-activity` and `month-calendar-heatmap` are timezone-sensitive by design (WOOA7S-2111) but not in the `test-tz` list in `package.json`. They pass under all three zones above, so adding them is free.

File: projects/packages/premium-analytics/packages/widgets-toolkit/src/components/month-calendar-heatmap/month-calendar-heatmap.tsx (hunk 1 above)
Comment (chihsuan) [reply projects/packages/premium-analytics/packages/widgets-toolkit/src/components/month-calendar-heatmap/month-calendar-heatmap.tsx:84]: Good catch, switched to calendar navigation and added the boundary test in 57d3a9c.

File: projects/packages/premium-analytics/packages/widgets-toolkit/src/helpers/calendar-heatmap-window.ts (hunk 2 above)
Comment (chihsuan) [reply projects/packages/premium-analytics/packages/widgets-toolkit/src/helpers/calendar-heatmap-window.ts:123]: Done, removed in de0534b.

**Comment by chihsuan** [comment]
Thanks @louwie17 and @dognose24 for the thorough testing! Both blocking items are in, plus the scroll suggestion, so the grid now opens on the current month. Tests are tidied up too. On the title, the current wording matches the prototype, so I'd rather revisit that later as a product call. Could you take another look?

**Review by louwie17 — APPROVED** [review]
Thanks for the updates @chihsuan! The keyboard now steps day by day across the months and the grid opens on the current month, this tested well on a local Docker site and code looks good.

Just a thought, not a blocker: when the grid opens scrolled to the current month, the first arrow key jumps back to Oct 1. Curious if starting on today would feel more natural there, but happy to leave that as is.

LGTM 🚀

**Review by dognose24 — APPROVED** [review]
Re-checked at `8641223`. Calendar keyboard navigation with the boundary test, the dead-code removal, the open-on-current-month scroll (the responsive wrapper mounts the grid on first render, so the layout effect sees it), the `test-tz` additions and the test cleanup all look good; CI is green. Leaving the title wording to product as you suggested. Thanks! 🚀
